src/components/SearchableMultiSelect/SearchableMultiSelect.jsx

```diff
@@ -407,39 +511,105 @@ const SearchableMultiSelect = createClass({
 								<LoadingIcon />
 							</DropMenu.Option>
 						: null}
-					{this.renderOptions(optionsProps)}
+					{this.renderOptions()}
 				</DropMenu>
 
 				{hasSelections && !_.isEmpty(selectedIndices)
 					? <Selection
 							isBold
 							hasBackground
-							Label="Selected"
 							kind="container"
 							onRemove={this.handleRemoveAll}
 							responsiveMode={responsiveMode}
 							isRemovable={hasRemoveAll}
 						>
+							<Selection.Label>
+								{selectionLabel.props.children}
+							</Selection.Label>
+							{_.map(
+								optionGroupDataLookup,
+								(groupedOptionsData, optionGroupIndex) => {
+									const selectedGroupedOptions = _.filter(groupedOptionsData, ({
+										optionIndex,
+									}) => _.includes(selectedIndices, optionIndex));
+									if (!_.isEmpty(selectedGroupedOptions)) {
+										return (
+											<Selection
+												key={'optionGroup-' + optionGroupIndex}
+												responsiveMode={responsiveMode}
+												isRemovable={false}
+												isBold
+												kind="container"
+											>
+												<Selection.Label>
+													{_.first(
+														rejectTypes(
+															optionGroups[optionGroupIndex].children,
+															SearchableMultiSelect.Option
+														)
+													)}
+												</Selection.Label>
+												{_.map(selectedGroupedOptions, ({
+													optionIndex,
+													optionProps,
+												}) => {
+													const selectionProps = _.get(
+														getFirst(
+															optionProps,
+															SearchableMultiSelect.Option.Selection
+														),
+														'props'
+													);
+													return (
+														<Selection
+															key={optionIndex}
+															{...selectionProps}
+															callbackId={optionIndex}
+															responsiveMode={responsiveMode}
+															onRemove={this.handleSelectionRemove}
+														>
+															<Selection.Label>
+																{optionProps.children}
+															</Selection.Label>
+														</Selection>
+													);
+												})}
+											</Selection>
+										);
+									}
+									return null;
+								}
+							)}
 							{_.map(selectedIndices, selectedIndex => {
-								const optionProps = optionsProps[selectedIndex];
-								const selectionProps = _.get(
-									getFirst(optionProps, SearchableMultiSelect.Option.Selection),
-									'props'
+								const selectedUngroupedOptionData = _.find(
+									ungroupedOptionData,
+									({ optionIndex }) => selectedIndex === optionIndex
```

Comment:
/nit `_.find(ungroupedOptionData, {optionIndex: selectedIndex})` shorthand is slightly more concise